### ISSUE TITLE
Upgrade to retry-as-promised 6.1.0

### ISSUE THREAD

**Issue by dylanlingelbach** [sequelize/sequelize#14952]
## Issue Creation Checklist

- [x] I understand that my issue will be automatically closed if I don't fill in the requested information
- [x] I have read the [contribution guidelines](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md)

## Bug Description

`retry-as-promised` was recently bumped to 6.1.0 to fix https://github.com/mickhansen/retry-as-promised/issues/31.

That issue was found as we upgraded from Sequelize v4 to v6.

### Reproducible Example

See https://github.com/mickhansen/retry-as-promised/issues/31. Add a `report` function to `retry` config that looks for an `err` object

### What do you expect to happen?

`retry.report` to receive the error object that is being retried

### What is actually happening?

`retry.report` does not receive the error object that is being retried

### Environment

- Sequelize version: 6.21.4
- Node.js version: v12.22.12
- If TypeScript related: TypeScript version: N/A
- Database & Version: N/A
- Connector library & Version: N/A

## Would you be willing to resolve this issue by submitting a Pull Request?

- [ ] Yes, I have the time and I know how to start.
- [ ] Yes, I have the time but I will need guidance.
- [ ] No, I don't have the time, but my company or I are [supporting Sequelize through donations on OpenCollective](https://opencollective.com/sequelize).
- [x] No, I don't have the time, and I understand that I will need to wait until someone from the community or maintainers is interested in resolving my issue.

---

*Indicate your interest in the resolution of this issue by adding the 👍 reaction. Comments such as "+1" will be removed.*

**Comment by dylanlingelbach** [comment]
@fzn0x yes, the issue was our logic that logged retryable warnings failed because we no longer had access to the `err` object in the `report` function.

Upgrading to `6.1.0` will fix this issue

**Comment by fzn0x** [comment]
Thanks @dylanlingelbach :)

**Comment by dylanlingelbach** [comment]
Thank you for getting it in so quickly!
